### PR TITLE
[13.0] [MIG+REF] mass_editing

### PR DESCRIPTION
**Pull request by legalsylvain**
**mass_editing improvment** : 
- [x] renaming ``mass.object`` to ``mass.editing`` for consistency
- [x] make options (set / add / remove) translatable
- [x] simplify ``mass.editing.wizard`` (V12: 300 lines ; V13: 160 lines)
- [x] make easier and add modularity in the ``mass.editing.wizard`` python code.
- [x] replace field_ids M2M fields by a new model``mass.editing.line`` to have the possibility to order fields in the wizard, by a sequence and to select the widget for each field.

image

- [x] migration Script.

**Add abstraction** (``mass.operation.abstract``)
- [x] Introducing a new abstract module ``mass_operation_abstract`` to provide mixin objects that allow to make mass operation on many items of any models 
- [x] make ``mass_editing`` depending on ``mass_operation_abstract``

### CODE REVIEW

**Comment by legalsylvain** [comment]
ready for review. cc : @SerpentCS

**Comment by legalsylvain** [comment]
For your information others module will come depending on the module ``mass_operation_abstract``. like this one : ``mass_action`` to execute python code for many items. 

ref : https://github.com/grap/grap-odoo-incubator/tree/8.0/mass_action

File: mass_operation_abstract/wizard/mass_operation_wizard_mixin.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# Copyright (C) 2018 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import Warning as UserError
```
Comment (bealdav) [inline mass_operation_abstract/wizard/mass_operation_wizard_mixin.py]:
```suggestion
from odoo.exceptions import UserError
```

Comment (legalsylvain) [reply mass_operation_abstract/wizard/mass_operation_wizard_mixin.py]: @bealdav : done. thanks for the review. did you tested functionnaly on runbot ? 
regards.

**Comment by SerpentCS** [comment]
Looks good (y). should not you make another PR for second module mass_operation_abstract?

**Comment by legalsylvain** [comment]
> Looks good (y). 

Thanks ! Is it an approval ? ;-)

> should not you make another PR for second module mass_operation_abstract?

Well, I split the module mass_editing in two modules, with a dependency. So it makes no sense to make two separate PR as one can not be merged without the other.  And if spitted, it will be more complicated for reviewer to see what code moved in the abstract module.

**Comment by SerpentCS** [comment]
Its an approval yes!

**Review by SerpentCS — APPROVED** [review]

**Review by tarteo — COMMENTED** [review]
Is this module still relevant in 13.0 since there is multi_edit="1"?

**Comment by legalsylvain** [comment]
Hi @tarteo : Thanks for your review. Regarding your question, did you read the description of the new module ? I think it answers to your question in the section "Difference between the Odoo 13.0 CE Feature"

https://github.com/legalsylvain/server-ux/blob/13.0-MIG-mass_editing/mass_editing/readme/DESCRIPTION.rst

Let me know if you still have questions.

**Comment by legalsylvain** [comment]
hi @bealdav I took into account your remarks. Could you update your review ? 

Thanks !

**Review by bealdav — APPROVED** [review]
Thanks. Please merge all fixup commits

**Comment by legalsylvain** [comment]
Oca bot merge fixup commit. Doesnt he?

**Comment by bealdav** [comment]
No sorry

**Comment by MiquelRForgeFlow** [comment]
@legalsylvain please squash fixup commits

**Review by ivantodorovich — APPROVED** [review]
code & functional test :+1:

**Comment by legalsylvain** [comment]
I squashed all the commits. Thanks for your reviews.

**Comment by mowny** [comment]
After upgrading from 12 to 13, installing mass_operation_abstract and updating mass_editing, odoo still complains about:

`2020-08-25 13:11:23,009 6349 INFO live_20161130_v13_test5_20200824 odoo.modules.loading: Model mass.object is declared but cannot be loaded! (Perhaps a module was partially removed or renamed)`

As this is only INFO (and the module seems to work) it's probably not _that_ bad, but it is distracting and I'd like to get rid of it. I just don't know how, so any suggestions are welcome.

**Pull request merged**